src/ethernet/ip_device.cpp

```diff
@@ -0,0 +1,310 @@
+{
+    std::cout << "GetControls\n";
+    std::vector<IpDeviceControlData> controls;
+    controls = remote_sensors[sensor_id]->rtsp_client->getControls();
+
+    return controls;
+}
+
+bool ip_device::init_device_data(rs2::software_device sw_device)
+{
+    std::string url, sensor_name = "";
+    for (int sensor_id = 0; sensor_id < NUM_OF_SENSORS; sensor_id++)
+    {
+
+        url = std::string("rtsp://" + ip_address + ":8554/" + sensors_str[sensor_id]);
+        sensor_name = sensors_str[sensor_id] + std::string(" (Remote)");
+
+        remote_sensors[sensor_id] = new ip_sensor();
+
+        remote_sensors[sensor_id]->rtsp_client = RsRTSPClient::getRtspClient(url.c_str(), "ip_device_device");
+        ((RsRTSPClient *)remote_sensors[sensor_id]->rtsp_client)->initFunc(&rs_rtp_stream::get_memory_pool());
+
+        rs2::software_sensor tmp_sensor = sw_device.add_sensor(sensor_name);
+
+        remote_sensors[sensor_id]->sw_sensor = std::make_shared<rs2::software_sensor>(tmp_sensor);
+
+        //hard_coded
+        if (sensor_id == 1) //todo: remove hard
+        {
+            std::vector<IpDeviceControlData> controls = get_controls(sensor_id);
+            for (auto &control : controls)
+            {
+                remote_sensors[control.sensorId]->sw_sensor->add_option(control.option, {control.range.min, control.range.max, control.range.def, control.range.step});
+                remote_sensors[control.sensorId]->sensors_option[control.option] = control.range.def;
+            }
+        }
+
+        auto streams = query_streams(sensor_id);
+
+        std::cout << "\t@@@ got " << streams.size() << " streams per sensor " << sensor_id << std::endl;
+
+        for (int stream_index = 0; stream_index < streams.size(); stream_index++)
+        {
+            // just for readable code
+            rs2_video_stream st = streams[stream_index];
+
+            //todo: remove
+            st.intrinsics = st.intrinsics;
+            //nhershko: check why profile with type 0
+            long long int stream_key = RsRTSPClient::getStreamProfileUniqueKey(st);
+            streams_collection[stream_key] = std::make_shared<rs_rtp_stream>(st, remote_sensors[sensor_id]->sw_sensor->add_video_stream(st, stream_index == 0));
+            memory_pool = &rs_rtp_stream::get_memory_pool();
+        }
+        std::cout << "\t@@@ done adding streams for sensor ID: " << sensor_id << std::endl;
+    }
+
+    //poll sw device streaming state
+    this->sw_device_status_check = std::thread(&ip_device::polling_state_loop, this);
+    return true;
+}
+
+void ip_device::polling_state_loop()
+{
+    while (this->is_device_alive)
+    {
+        bool enabled;
+        for(int i=0 ; i < NUM_OF_SENSORS ; i++ )
+        {
+            //poll start/stop events
+            auto sw_sensor = remote_sensors[i]->sw_sensor.get();
+            //auto current_active_streams = sw_sensor->get_active_streams();
+            
+            if (sw_sensor->get_active_streams().size() > 0)
```

Comment:
can be `enabled = sw_sensor->get_active_streams().size() > 0;`

Reply:
done

Reply:
Done